File: 0001-two-sum/0001-two-sum.java
import java.util.*;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer>mp=new HashMap<>();
        for(int i=0;i<nums.length;i++)
        {
            int more=target-nums[i];
            if(mp.containsKey(more))
            {
                return new int[]{mp.get(more),i};
            }
            mp.put(nums[i],i);
        }
        return new int[]{};
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        seen = {}

        for i in range(len(nums)):
            more = target - nums[i]

            if more in seen:
                return [seen[more], i]

            seen[nums[i]] = i

        return []